### PR TITLE
JAMES-2666 solve group deserialization issue with event dead letters …

### PR DESCRIPTION
**Pull request by Arsnael**
…routes

### CODE REVIEW

**Comment by Arsnael** [comment]
In dead letters routes, when we try to deserialize the group argument and instance it into a `Group`, it fails as all our groups nested in our listeners were declared `private`. So this PR is a quick fix.

**Comment by trantienduchn** [comment]
I prefer to treat with privately accessing by java reflection rather than writing many tests for every group serialization and related listeners.

File: mailbox/caching/pom.xml [hunk 1]
```diff
@@ -54,6 +54,16 @@
             <artifactId>junit</artifactId>
```
Comment (chibenwa) [inline mailbox/caching/pom.xml]: Replace this dependency with junit-vintage

**Comment by chibenwa** [comment]
> I prefer to treat with privately accessing by java reflection rather than writing many tests for every group serialization and related listeners.

Well, do we agree that:

 - We need to be able to deserialize **every** group, hence we should have tests for it?
 - Reflection is over technical. And ugly. I'd be glad to never use that in tests
 - When testing we should never test the **how** but test the expected actions.

Note that rené will be adding a few "real world integration tests" replicating the "pause elasticSearch + reschedule stuff" demo scenario.

File: mailbox/caching/pom.xml (hunk 1 above)
Comment (Arsnael) [reply mailbox/caching/pom.xml]: I think I can just delete it in fact !

**Comment by trantienduchn** [comment]
> Reflection is over technical. And ugly. I'd be glad to never use that in tests

I didn't mean to use reflection it in tests. If i can, I will use it in the code (BTW, we already use it in the code) to access private constructors from out side and able to create new instance for every private group. And with this impl, I feel confident that just one more unit test is enough to me to show that Group deserialization can work with private classes.

**Comment by Arsnael** [comment]
> > Reflection is over technical. And ugly. I'd be glad to never use that in tests
> 
> I didn't mean to use reflection it in tests. If i can, I will use it in the code (BTW, we already use it in the code) to access private constructors from out side and able to create new instance for every private group. And with this impl, I feel confident that just one more unit test is enough to me to show that Group deserialization can work with private classes.

I didn't know much about Java reflection first so thanks to raise the thing, I learnt something. But I disagree. If something is declared `private`, it should stay like this and not be accessible from outside of the class it was declared. That solution kinda breaks the private contract I think !

**Comment by chibenwa** [comment]
> That solution kinda breaks the private contract I think !

:+1: 

I prefer the solution of saying 'implementers are responsible of writing deserializable groups'

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**